setup.py

```diff
@@ -1,16 +1,40 @@
 import os
 import re
 from pathlib import Path
+from typing import List
 
-# https://setuptools.pypa.io/en/latest/pkg_resources.html
-import pkg_resources  # noqa: TID251  # TODO: switch to poetry
 from setuptools import find_packages, setup
 
 import versioneer
 
 SUPPORTED_PYTHON = ">=3.9,<3.13"
 
 
+def parse_requirements(file_path: Path) -> List[str]:
```

Comment:
Does this match the implementation in pip? Can you provide a link?

Reply:
I found it and it looks more complicated unfortunately: https://github.com/pypa/pip/blob/main/src/pip/_internal/req/req_file.py#L146